src/schemas.js

```diff
@@ -171,3 +170,247 @@ export const atomicSwapCreateSchema = {
+        if ('type' in value && value.type === 'data') {
+          if ('data' in value && !('token' in value)) {
+            return (/^[0-9a-fA-F]+$/.test(value.data));
+          }
+          return false;
+        }
+        // This is an address output
+        // Check that we have all params
+        return ('address' in value && 'value' in value);
+      },
+    },
+  },
+  inputs: {
+    in: ['body'],
+    errorMessage: 'Invalid inputs array',
+    isArray: true,
+    notEmpty: true,
+    optional: true,
+  },
+  change_address: {
+    in: ['body'],
+    errorMessage: 'Invalid change address',
+    isString: true,
+    optional: true,
+  },
+};
+
+export const txInputSchema = {
+  'inputs.*.hash': {
+    in: ['body'],
+    errorMessage: 'Invalid hash',
+    isString: true,
+    custom: {
+      // Test if hash is a 64 character hex string
+      options: value => /^[0-9a-fA-F]{64}$/.test(value),
+    },
+  },
+  'inputs.*.index': {
+    in: ['body'],
+    errorMessage: 'Invalid index',
+    isInt: {
+      options: {
+        min: 0,
+      },
+    },
+    toInt: true,
+  },
+};
+
+export const queryInputSchema = {
+  inputs: {
+    in: ['body'],
+    errorMessage: 'Invalid inputs array',
+    isArray: true,
+    isLength: 1,
+    notEmpty: true,
+  },
+  'inputs.*.type': {
+    in: ['body'],
+    errorMessage: 'Invalid type',
+    isString: true,
+    equals: {
+      options: 'query',
+    },
+  },
+  'inputs.*.max_utxos': {
+    in: ['body'],
+    errorMessage: 'Invalid max_utxos',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.filter_address': {
+    in: ['body'],
+    errorMessage: 'Invalid filter_address',
+    isString: true,
+    optional: true,
+  },
+  'inputs.*.amount_smaller_than': {
+    in: ['body'],
+    errorMessage: 'Invalid amount_smaller_than',
+    isInt: {
+      options: {
+        min: 2,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.amount_bigger_than': {
+    in: ['body'],
+    errorMessage: 'Invalid amount_bigger_than',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+  'inputs.*.maximum_amount': {
+    in: ['body'],
+    errorMessage: 'Invalid maximum_amount',
+    isInt: {
+      options: {
+        min: 1,
+      },
+    },
+    toInt: true,
+    optional: true,
+  },
+};
+
+export const txHexInputDataSchema = {
+  ...txHexSchema,
+  signatures: {
+    in: ['body'],
+    errorMessage: 'Invalid signatures',
+    isArray: true,
+    isEmpty: false,
+  },
+  'signatures.*.index': {
+    in: ['body'],
+    errorMessage: 'Invalid signature input index',
+    isInt: {
+      options: {
+        min: 0,
+      },
+    },
+    toInt: true,
+  },
+  'signatures.*.data': {
+    in: ['body'],
+    errorMessage: 'Invalid signature data',
+    isString: true,
+    custom: {
+      options: value => /^[0-9a-fA-F]+$/.test(value),
```

Comment:
We have this hex validation test in many places here. Maybe just create a method at the beginning of the file to reuse this. What do you think?

Reply:
The hex validation is slightly different in some cases.
Where the length is unknown we just test for hex values, but for tx_ids (i.e. sha256, 64 hex chars) we test the length.
And if it's a token uid we may accept HTR `00` as well.

If we were to have a specific validation method for token uids and signatures maybe it should come from the wallet-lib.
What do you think?

Reply:
Yeah, makes sense. It's fine